src/AssetPublisher.php

```diff
@@ -123,21 +257,20 @@ public function loadBundle(AssetManager $am, string $name, array $config = []):
      * @throws InvalidConfigException if the asset to be published does not exist.
      *
      */
-    public function publish(AssetManager $am, AssetBundle $bundle): array
+    public function publish(AssetBundle $bundle): array
     {
-        $this->checkBasePath($am, $bundle->basePath);
-        $this->checkBaseUrl($am, $bundle->baseUrl);
-
         if (isset($this->published[$bundle->sourcePath])) {
             return $this->published[$bundle->sourcePath];
         }
 
-        if (!file_exists($am->getAliases()->get($bundle->sourcePath))) {
+        if (!file_exists($this->aliases->get($bundle->sourcePath))) {
             throw new InvalidConfigException("The sourcePath to be published does not exist: $bundle->sourcePath");
         }
 
+        $this->checkBasePath($bundle->basePath);
+        $this->checkBaseUrl($bundle->baseUrl);
+
         return $this->published[$bundle->sourcePath] = $this->publishDirectory(
-            $am,
             $bundle->sourcePath,
```

Comment:
`sourcePath` could be null

Reply:
As always published from the AssetManager first make the check in AssetPublisher::loadBundle, and if you use the publisher directly there if it could happen, place another check.

Reply:
file_exists('null') its false,

https://github.com/terabytesoftw/assets/blob/73c37cd03efd280a5aef7819007e0f4092c496cd/src/AssetPublisher.php#L266

Reply:
I don't think we should trigger file system check if it's null.

Reply:
https://github.com/terabytesoftw/assets/blob/73c37cd03efd280a5aef7819007e0f4092c496cd/src/AssetPublisher.php#L270

Reply:
But these aren't checking `sourcePath`.

Reply:
done.